tests/unit/s2n_mem_allocator_test.c

```diff
@@ -189,15 +184,22 @@ int main(int argc, char **argv)
             mock_client(&io_pair);
         }
 
+        EXPECT_NOT_NULL(cert_chain_pem = malloc(S2N_MAX_TEST_PEM_SIZE));
```

Comment:
I'm assuming there is a reason that we don't use DEFER_CLEANUP here. Does something between here and the call to free fork()? Or am I misunderstanding?

Reply:
This is special case since we are running multiple test within a `for` loop.

The tenets I tried to follow were (and some of them conflict):
- keep change to minimum
- avoid extra calls to free (move init/alloc after call to fork)
- simplify where possible (use DEFER when it made sense)